src/views/FindFacilities.vue

```diff
@@ -260,7 +261,58 @@ export default defineComponent({
         showToast(translate('Failed to update fulfillment capacity for ', { facilityName: facility.facilityName }))
         logger.error('Failed to update facility', err)
       }
-    }
+    },
+    async updateSellOnlineStatus(facility: any) {
+      if (!facility.sellOnline) {
+        this.addFacilityToGroup(facility.facilityId)
+      } else {
+        this.updateFacilityToGroup(facility)
+      }
+    },
+    async addFacilityToGroup(facilityId: string) {
```

Comment:
Instead of writing these two separate functions, addFacilityToGroup and updateFacilityToGroup, incorporate the logic into the updateSellOnlineStatus function itself. This will reduce the code size, as we have the same error handling. The code will still be readable, as we have independent services for the same.